test/lib/ansible_test/_internal/commands/sanity/validate_modules.py

```diff
@@ -159,7 +159,11 @@ def test(self, args: SanityConfig, targets: SanityTargets, python: PythonConfig)
                 temp_dir = process_scoped_temporary_directory(args)
 
                 with tarfile.open(path) as file:
-                    file.extractall(temp_dir)
+                    if hasattr(tarfile, 'data_filter'):
+                        # Remove this check when Python 3.11 is EOL
+                        file.extractall(temp_dir, filter='data')
```

Comment:
Suppress the `mypy` type error since it doesn't understand the `hasattr` guard we're using:

```suggestion
                        file.extractall(temp_dir, filter='data')  # type: ignore[call-arg]
```